src/main/java/com/ne4istb/test/geofences/GeofencingService.java

```diff
@@ -5,130 +5,133 @@
-                break;
-            case REMOVE:
-                Log.d("GEO", "Location client removes geofence");
-                mLocationClient.removeGeofences(mGeofenceListsToRemove, this);
-                break;
-        }
-    }
-
-    private PendingIntent getPendingIntent() {
-        Intent transitionService = new Intent(this, ReceiveTransitionsIntentService.class);
-        return PendingIntent.getService(this, 0, transitionService, PendingIntent.FLAG_UPDATE_CURRENT);
-    }
-
-    @Override
-    public void onDisconnected() {
-        Log.d("GEO", "Location client disconnected");
-    }
-
-    @Override
-    public void onAddGeofencesResult(int i, String[] strings) {
-        if (LocationStatusCodes.SUCCESS == i) {
-
-            Log.d("GEO", "Geofences added " + strings);
-
-            for (String geofenceId : strings)
-                Toast.makeText(this, "Geofences added: " + geofenceId, Toast.LENGTH_SHORT).show();
-
-            mLocationClient.disconnect();
-            stopSelf();
-        } else {
-            Log.e("GEO", "Error while adding geofence: " + strings);
-        }
-    }
-
-    @Override
-    public void onRemoveGeofencesByRequestIdsResult(int i, String[] strings) {
-        if (LocationStatusCodes.SUCCESS == i) {
-            Log.d("GEO", "Geofences removed" + strings);
-            mLocationClient.disconnect();
-            stopSelf();
-        } else {
-            Log.e("GEO", "Error while removing geofence: " + strings);
-        }
-    }
-
-    @Override
-    public void onRemoveGeofencesByPendingIntentResult(int i, PendingIntent pendingIntent) {
-    }
-
-    @Override
-    public void onConnectionFailed(ConnectionResult connectionResult) {
-        Log.e("GEO", "Location client connection failed: " + connectionResult.getErrorCode());
-    }
-
-    @Override
-    public IBinder onBind(Intent intent) {
-        return null;
-    }
-
-    @Override
-    public void onDestroy() {
-        Log.d("GEO", "Location service destroyed");
-        super.onDestroy();
-    }
+public class GeofencingService extends Service implements GoogleApiClient.ConnectionCallbacks,
+		GoogleApiClient.OnConnectionFailedListener {
+
+	public static final String EXTRA_REQUEST_IDS = "requestId";
+	public static final String EXTRA_GEOFENCE = "geofence";
+	public static final String EXTRA_ACTION = "action";
+
+	private List<Geofence> mGeofenceListsToAdd = new ArrayList<>();
+	private GoogleApiClient mGoogleApiClient;
+	private Action mAction;
+	private int transitionType;
+
+	@Override
+	public int onStartCommand(Intent intent, int flags, int startId) {
+
+		Log.d("GEO", "Location service started");
+
+		mAction = (Action) intent.getSerializableExtra(EXTRA_ACTION);
+
+		switch (mAction) {
+			case ADD:
+				MyGeofence newGeofence = (MyGeofence) intent.getSerializableExtra(EXTRA_GEOFENCE);
+				transitionType = newGeofence.getTransitionType();
+				mGeofenceListsToAdd.add(newGeofence.toGeofence());
+				break;
+			case REMOVE:

```

Comment:
But i'm not sure if we should not remove geofences. Previously, they were kept until phone reboot. Any similar problems with new API?